### PR TITLE
meson: enable more compiler warnings

### PR DESCRIPTION
**Pull request by emersion**

### CODE REVIEW

**Comment by agx** [comment]
Looks good to me!

File: meson.build [hunk 1]
```diff
@@ -16,16 +16,31 @@ project(
...
+	'-Wredundant-decls',
+	'-Wimplicit-fallthrough=2',
+	'-Wendif-labels',
+	'-Wstrict-aliasing=2',
+	'-Woverflow',
+
+	'-Wno-missing-braces',
+	'-Wno-missing-field-initializers',
+	'-Wno-unused-parameter',
+]), language: 'c')
```
Comment (ddevault) [inline meson.build]: Can we summarize this as `-Wall -Wextra -Wpedantic`?

Comment (emersion) [reply meson.build]: No, `-Wpedantic` gives a lot of these:

```
../types/wlr_output_layout.c: In function ‘wlr_output_layout_add’:
../types/wlr_output_layout.c:211:66: warning: ISO C99 requires at least one argument for the "..." in a variadic macro
    wlr_log(WLR_ERROR, "Failed to create wlr_output_layout_output");
```

And there's no way to disable this specific warning. There's also no easy way to fix `wlr_log`.

Comment (ascent12) [reply meson.build]: Also, meson already control those as as `warning_level` option.

Comment (emersion) [reply meson.build]: Yeah, the idea was to bump `warning_level` to 3 to enable `-Wpedantic`

**Pull request merged**